app/models/test_result.rb

```diff
@@ -16,6 +16,9 @@ class TestResult < ApplicationRecord
   validates :end_time, presence: true,
                        uniqueness: { scope: %i[host_id profile_id] }
 
+  after_save :update_profile_score!
+  after_destroy :update_profile_score!
```

Comment:
Please check if this would not have a high impact on (bulk) Host deletion triggering a TestResult cleanup.

Reply:
Well, I'm a bit worried that it will cause an `N-1` but I have to think a bit more about how to avoid that :confused:

Reply:
Okay, so the [`delete_all`](https://github.com/RedHatInsights/compliance-backend/blob/master/app/jobs/delete_host.rb#L28) method here is skipping all the callbacks [according to the docs](https://guides.rubyonrails.org/active_record_callbacks.html#skipping-callbacks) so this should be safe as it is.

Reply:
Wouldn't that leave the old score cached / not recalculated?

Reply:
True, so what about building a set of affected profile IDs in the `remove_related` method and batch-updating their score?

Reply:
> True, so what about building a set of affected profile IDs in the `remove_related` method and batch-updating their score?

That sound great!

Reply:
Sounds good, doesn't work :laughing: at least it might introduce useless computation. Depends on how many hosts are we deleting at once? :thinking:

Reply:
Keep in mind the DeleteHostsJob. It uses `delete` and not `destroy`, so the score would need to be updated if the host had any test results in Compliance. An alternative approach would be to just 'invalidate' the score by setting it to `null` and only re-calculating it when it's requested. I'm not suggesting we do this, just suggesting we think about if it would be a better solution :)

Reply:
@akofink already solved that, but I'm concerned about the performance when deleting a lot of hosts :thinking: